src/server/plugins/engine/components/YesNoField.test.ts

```diff
@@ -1,104 +1,190 @@
-      name: 'speakEnglish',
+  const definition = {
+    pages: [],
+    lists: [],
+    sections: [],
+    conditions: []
+  } satisfies FormDefinition
+
+  let def: YesNoFieldComponent
+  let formModel: FormModel
+  let component: YesNoField
+  let label: string
+  let allow: boolean[]
+
+  beforeEach(() => {
+    def = {
+      title: 'Example yes/no',
+      name: 'myComponent',
       type: ComponentType.YesNoField,
       options: {},
       schema: {}
-    }
-
-    const formModel = {
-      makePage: () => jest.fn(),
-      getList: () => ({
-        name: '__yesNo',
-        title: 'Yes/No',
-        type: 'boolean',
-        items: [
-          {
-            text: 'Yes',
-            value: true
-          },
-          {
-            text: 'No',
-            value: false
-          }
-        ]
-      })
-    }
-
-    describe('getViewModel', () => {
-      it('viewModel item Yes is checked when evaluating boolean true', () => {
-        const component = new YesNoField(componentDefinition, formModel)
-        const payload = {
-          speakEnglish: true
-        }
-
-        const viewModel = component.getViewModel(payload)
-        const yesItem = viewModel.items?.filter(
-          (item) => item.text === 'Yes'
-        )[0]
-
-        expect(yesItem).toEqual({
-          text: 'Yes',
-          value: 'true',
-          checked: true
+    } satisfies YesNoFieldComponent
+
+    formModel = new FormModel(definition, {
+      basePath: 'test'
+    })
+
+    component = new YesNoField(def, formModel)
+    label = def.title.toLowerCase()
+    allow = [true, false]
+  })
+
+  describe('Schema', () => {
+    it('uses component title as label', () => {
+      const { formSchema } = component
+
+      expect(formSchema.describe().flags).toEqual(
+        expect.objectContaining({ label })
+      )
+    })
+
+    it('is required by default', () => {
+      const { formSchema } = component
+
+      expect(formSchema.describe().flags).toEqual(
+        expect.objectContaining({
+          presence: 'required'
         })
-      })
+      )
+    })
+
+    it('is optional when configured', () => {
+      const componentOptional = new YesNoField(
+        { ...def, options: { required: false } },
+        formModel
+      )
 
-      it("viewModel item Yes is checked when evaluating string 'true'", () => {
-        const component = new YesNoField(componentDefinition, formModel)
-        const payload = {
-          speakEnglish: 'true'
-        }
-
-        const viewModel = component.getViewModel(payload)
-        const yesItem = viewModel.items?.filter(
-          (item) => item.text === 'Yes'
-        )[0]
-
-        expect(yesItem).toEqual({
-          text: 'Yes',
-          value: 'true',
-          checked: true
+      const { formSchema } = componentOptional
+
+      expect(formSchema.describe().flags).not.toEqual(
+        expect.objectContaining({
+          presence: 'required'
         })
-      })
+      )
```

Comment:
[In PR #227](https://github.com/DEFRA/forms-runner/pull/227/files#diff-8803b282b08e06522864d16b8f9289639e904d18f308216f017ad924857ee671L84), you use a check for 'optional' rather than 'not required'. Worth keeping them consistent? Not a blocker.

Reply:
Yeah this one's "not required" for consistency (it merges first)

But then in #277 I'm flipping them all from "not required" to clearly be "optional"

Sorry, my fault for stacking the commits and working on this one first

Reply:
Excellent spot though, you passed 💯